app/src/marketplace/containers/ProductController/useUserHasAccessToProject.ts

```diff
@@ -1,16 +1,18 @@
-import {
-    hasActiveProjectSubscription,
-    isPaidProject,
-    isProjectOwnedBy,
-} from '$mp/utils/product'
+import { isPaidProject, isProjectOwnedBy } from '$mp/utils/product'
 import { useLoadedProject } from '$mp/contexts/LoadedProjectContext'
 import { useAuthController } from '$app/src/auth/hooks/useAuthController'
+import { useHasActiveProjectSubscription } from '$shared/stores/purchases'
 
 export const useUserHasAccessToProject = (): boolean => {
     const { loadedProject: project, theGraphProject } = useLoadedProject()
 
     const { currentAuthSession: { address = undefined } = {} } = useAuthController() || {}
 
+    const hasActiveProjectSubscription = useHasActiveProjectSubscription(
```

Comment:
`useHasActiveProjectSubscription` is smart enough to load current subscriptions, reuse, and re-fetch if the state got invalidated.